tests/common/include/tests/common/tools_line_intersection.inl

```diff
@@ -0,0 +1,157 @@
+    is[0] = ray_line_intersector().intersect(tf, trks[0], ln);
+    is[1] = ray_line_intersector().intersect(tf, trks[1], ln);
+
+    EXPECT_EQ(is[0].status, intersection::status::e_inside);
+    EXPECT_EQ(is[0].path, 1);
+    EXPECT_EQ(is[0].p3, point3({1, 0, 0}));
+    EXPECT_EQ(is[0].p2, point2({1, 0}));
+
+    EXPECT_EQ(is[1].status, intersection::status::e_inside);
+    EXPECT_EQ(is[1].path, 1);
+    EXPECT_EQ(is[1].p3, point3({-1, 0, 0}));
+    EXPECT_EQ(is[1].p2, point2({1, 0}));
+}
+
+TEST(tools, line_intersector_square_scope) {
+
+    // tf3 with Identity rotation and no translation
+    vector3 x{1, 0, 0};
+    vector3 y{0, 1, 0};
+    vector3 z{0, 0, 1};
+    vector3 t{0, 0, 0};
+
+    const transform3 tf{t, x, y, z};
+
+    /// Create a track
+    std::vector<free_track_parameters> trks;
+    trks.push_back({{2, 0, 0}, 0, {-1, 1, 0}, -1});
+    trks.push_back({{1.9, 0, 0}, 0, {-1, 1, 0}, -1});
+    trks.push_back({{2.1, 0, 0}, 0, {-1, 1, 0}, -1});
+
+    trks.push_back({{-2, 0, 0}, 0, {1, 1, 0}, -1});
+    trks.push_back({{-1.9, 0, 0}, 0, {1, 1, 0}, -1});
+    trks.push_back({{-2.1, 0, 0}, 0, {1, 1, 0}, -1});
+
+    trks.push_back({{0, -2, 0}, 0, {-1, 1, 0}, -1});
+    trks.push_back({{0, -1.9, 0}, 0, {-1, 1, 0}, -1});
+    trks.push_back({{0, -2.1, 0}, 0, {-1, 1, 0}, -1});
+
+    trks.push_back({{0, -2, 0}, 0, {1, 1, 0}, -1});
+    trks.push_back({{0, -1.9, 0}, 0, {1, 1, 0}, -1});
+    trks.push_back({{0, -2.1, 0}, 0, {1, 1, 0}, -1});
+
+    // Infinite wire with 1 mm square cell size
+    line<ray_line_intersector, cartesian, dindex, true> ln{
+        1., std::numeric_limits<scalar>::infinity(), 0u};
+
+    // Test intersect
+    std::vector<ray_line_intersector::intersection_type> is;
+    for (const auto& trk : trks) {
+        is.push_back(
+            ray_line_intersector().intersect(tf, trk, ln, {1e-5, 1e-5}));
+    }
+
+    EXPECT_EQ(is[0].status, intersection::status::e_inside);
+    EXPECT_NEAR(is[0].path, sqrt(2), tolerance);
+    EXPECT_NEAR(is[0].p3[0], 1, tolerance);
+    EXPECT_NEAR(is[0].p3[1], 1, tolerance);
+    EXPECT_NEAR(is[0].p3[2], 0, tolerance);
+    EXPECT_NEAR(is[0].p2[0], sqrt(2), tolerance);
+    EXPECT_NEAR(is[0].p2[1], 0, tolerance);
+
+    EXPECT_EQ(is[1].status, intersection::status::e_inside);
+    EXPECT_EQ(is[2].status, intersection::status::e_outside);
+
+    EXPECT_EQ(is[3].status, intersection::status::e_inside);
+    EXPECT_EQ(is[4].status, intersection::status::e_inside);
+    EXPECT_EQ(is[5].status, intersection::status::e_outside);
+
+    EXPECT_EQ(is[6].status, intersection::status::e_inside);
+    EXPECT_EQ(is[7].status, intersection::status::e_inside);
+    EXPECT_EQ(is[8].status, intersection::status::e_outside);
+
+    EXPECT_EQ(is[9].status, intersection::status::e_inside);
+    EXPECT_EQ(is[10].status, intersection::status::e_inside);
+    EXPECT_EQ(is[11].status, intersection::status::e_outside);
+}
+
+TEST(tools, line_intersector_stereo_wire) {
```

Comment:
Sorry if I have to ask, but how does the modeling of a stereo wire work?

Reply:
Actually it's not right to call it stereo wire here. I changed to inclined_wire